pkg/webhook/mutation.go

```diff
@@ -159,3 +174,43 @@ func AppendMutationWebhookIfEnabled(webhooks []rotator.WebhookInfo) []rotator.We
 	}
 	return webhooks
 }
+
+// validateGatekeeperResources returns whether an issue is user error (vs internal) and any errors
+// validating internal resources
+func (h *mutationHandler) validateGatekeeperResources(ctx context.Context, req admission.Request) (bool, error) {
+	if req.AdmissionRequest.Kind.Group == mutationsGroup && req.AdmissionRequest.Kind.Kind == "AssignMetadata" {
+		return h.validateAssignMetadata(ctx, req)
+	}
+	if req.AdmissionRequest.Kind.Group == mutationsGroup && req.AdmissionRequest.Kind.Kind == "Assign" {
+		return h.validateAssign(ctx, req)
+	}
+	return false, nil
+}
+
+func (h *mutationHandler) validateAssignMetadata(ctx context.Context, req admission.Request) (bool, error) {
+	obj, _, err := deserializer.Decode(req.AdmissionRequest.Object.Raw, nil, &mutationsv1alpha1.AssignMetadata{})
+	if err != nil {
+		return false, err
```

Comment:
It might be user error if we can't deserialize the resource into the appropriate object.

Then again, that should have been caught by the API server checking the schema of the CRD, so maybe this is the appropriate way to handle things.

TL;DR I can see this both ways, so fine as-is